Add support for network device limits.priority option (#12135)
canonical/lxd: doc/reference/devices_nic.md

`limits.egress`         | string  | -                 | I/O limit in bit/s for outgoing traffic (various suffixes supported, see {ref}`instances-limit-units`)
`limits.ingress`        | string  | -                 | I/O limit in bit/s for incoming traffic (various suffixes supported, see {ref}`instances-limit-units`)
`limits.max`            | string  | -                 | I/O limit in bit/s for both incoming and outgoing traffic (same as setting both `limits.ingress` and `limits.egress`)

> `limits.priority`       | integer | -                 | `skb->priority` for outgoing traffic

Reviewer: I think we will need to clarify what this config item does in more accessible style, as I doubt many of our users will know what "skb->priority" is. Instead can you expand this to explain what it does as an end result, and whether the priority increases with a higher or lower number? We should also include the range of allowed integers here.

Reviewer: @mihalicyn this still needs addressing.

Author: I've updated description. Thanks!

> `limits.priority`       | integer | -                 | The `skb->priority` value (32-bit unsigned integer) for outgoing traffic, to be used by the kernel queuing discipline (qdisc) to prioritize network packets (The effect of this value depends on the particular qdisc implementation, for example, `SKBPRIO` or `QFQ`. Consult the kernel qdisc documentation before setting this value.)

Reviewer: This needs to be added to each relevant NIC

Author: Fixed. Thanks!

> `limits.priority`       | integer | -                 | `skb->priority` (32-bit unsigned integer) for outgoing traffic. This value is used by the kernel queueing discipline (qdisc) to prioritize network packets. Effect of this value depending on a particular qdisc implementation (`SKBPRIO`, `QFQ`, etc). You need to consult with the kernel qdisc documentation before setting anything here.

Reviewer 2: ```suggestion
`limits.priority`       | integer | -                 | The `skb->priority` value (32-bit unsigned integer) for outgoing traffic, to be used by the kernel queuing discipline (qdisc) to prioritize network packets (The effect of this value depends on the particular qdisc implementation, for example, `SKBPRIO` or `QFQ`. Consult the kernel qdisc documentation before setting this value.)
```
Trying to fit this into the existing format - this will work much better when we move this to the code and have more space for explanations. :)

Author: Thanks a lot! Fixed!
